great_expectations/rule_based_profiler/data_assistant/data_assistant_dispatcher.py

```diff
@@ -105,3 +105,16 @@ def get_data_assistant_impl(
             return None
         name = name.lower()
         return cls._registered_data_assistants.get(name)
+
+    def __dir__(self):
+        """
+        This custom magic method is used to enable expectation tab completion on "DataAssistantDispatcher" objects.
+        """
+        data_assistant_dispatcher_attrs: Set[str] = set(super().__dir__())
+        data_assistant_registered_names: Set[str] = set(
+            DataAssistantDispatcher._registered_data_assistants.keys()
```

Comment:
I very much agree - the `registered_data_assistants` dict was a global variable before but was refactored to be encapsulated within the `Dispatcher`.

Some sort of interface to interact with that private class attr would be preferred if possible 🙇🏽

Reply:
@jcampbell @cdkini I created the `get_registered_data_assistant_names()` method.  I believe that this achieves the compromise of moving the `DataAssistant` registry into the `DataAssistantDispatcher` class (per @cdkini ) and also using the `get_registered_data_assistant_names()` method, instead of accessing the private registry (per @jcampbell ).  Thanks.

Reply:
@jcampbell Per conversation with @donaldheppner and pod during the standup at noon Pacific Time on 5/10/2022, it was deemed as acceptable to merge in the present form (any refinement ideas are welcome and can be incorporated) as part of a future work.  Thank you.